lib/tasks/lumen.rake

```diff
@@ -741,6 +741,47 @@ where works.id in (
   # they were removed upon removal of the BlogEntry model, after having been
   # run to migrate production content. They were last available in commit
   # b45018d.
+
+  desc 'Run catchup ES indexing'
+  task run_catchup_es_indexing: :environment do
+    reindexing_timestamp_file = Rails.root.join('tmp', 'reindexing_timestamp')
+    log_message_prefix = '[rake lumen:run_catchup_es_indexing]'
+
+    unless File.exist?(reindexing_timestamp_file)
+      m = "#{log_message_prefix} tmp/reindexing_timestamp file doesn't exist"
+      puts m
+      Rails.logger.error m
+      exit
+    end
+
+    reindexing_start_date = File.mtime(reindexing_timestamp_file)
```

Comment:
Elegant.

Reply:
Although: what happens if we forget to create/touch this file at the time that we initiate the first indexing process? Can we retroactively edit (= lie about) mtime? Or are we going to need something that is less susceptible to our having possibly missed a step in the deploy (e.g. putting a notice ID or serialized DateTime object into the file and reading it)?

Reply:
I don't think that `putting a notice ID or serialized DateTime object into the file and reading it` is less susceptible. I we can forget to touch this file we can also to forget to insert a notice ID inside this file.

What we should probably do is to create a reindexing task (or update it if we have one) and make sure it creates the `reindexing_timestamp_file` file there.

Reply:
I also believe in our ability to forget to put a Notice ID in this file ;) However, if we have forgotten to do so the task can fail, and then we can go back in future and edit it and try again -- whereas if we've forgotten to touch the file I don't think we have a retrospective action there.

There is an index_changed_model_instances task which runs on a cron job and it could be set to touch a file, but this is going to be different from the massive reindexing job we do to move from ES 5 to ES 7. I think putting it into the reindexing task is better than relying on humans but I would still have the same question -- if we somehow fail to create the file, or we accidentally delete it, what's our recovery plan?

Reply:
> what's our recovery plan?

We will know the date of the reindexing process start and just update the timestamp of the file if something goes wrong. Just KISS it.

Reply:
Ah. OK. The missing factor here was that I was not aware you could set file mtimes to arbitrary numbers (including lies), and it proved shockingly hard to google for when I was trying to figure that out on first review. But given that in fact you can retroactively change them, this is fine.